src/uart/mod.rs

```diff
@@ -69,24 +112,57 @@ pub enum CharacterLength {
 /// The clock source to use for UART
 #[repr(u8)]
 pub enum ClockSources {
-    PLCK = 0,
+    PCLK = 0,
     IBRO = 2,
 }
 
 /// # Stop Bits
 /// The number of stop bits to use.
 /// Note: When using a character length of five bits, passing the variant
 /// `TwoBits` uses 1.5 bits.
-#[repr(u8)]
 pub enum StopBits {
-    OneBit = 0,
-    TwoBits = 1,
+    OneBit,
+    TwoBits,
+}
+
+/// # Threshold Size
+/// Then number of bits to apply to a FIFO threshold
+#[repr(u8)]
+pub enum ThresholdSize {
+    Threshold1 = 1,
+    Threshold2 = 2,
+    Threshold3 = 3,
+    Threshold4 = 4,
+    Threshold5 = 5,
+    Threshold6 = 6,
+    Threshold7 = 7,
+    Threshold8 = 8,
+}
+
+/// # Hardware Flow Control Deassert Condition
+/// When to deassert the hardware flow control
+pub enum HFCDeassertCondition {
+    EqualsFIFODepth,
+    ExceedsRxThreshold,
+}
+
+pub enum ParityValueSelect {
+    OneBased,
+    ZeroBased,
 }
 
 impl<Port: private::UARTPortCompatable> UART<Port> {
-    /// Creates an UART instance configured to use 8 bits, 1 stop bit, and no parity
-    /// TODO make this more generic
-    fn init() -> Self {
+    /// #
+    /// Creates a UART communication channel
+
+    fn init(
```

Comment:
`init` does not need a doc comment since its private to this file. The `port_*_init` should have the comments instead.